docs/path-resolution.rst

```diff
@@ -0,0 +1,690 @@
+               },
+               "/usr/local/dapp-bin/token.sol": {
+                   "content": "contract Token {}"
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   The ``sources`` dictionary specifies the initial content of the virtual filesystem and you
+   can use source unit names directly there.
+   They do not undergo any extra translation or normalization.
+
+   The path to the JSON file does not affect the path resolution in any way.
+   In fact, it is common to supply it on the standard input in which case it does not have a path at all.
+
+   .. note::
+
+       When using ``--standard-json`` you cannot provide additional source files as command-line
+       arguments but it does not mean that the compiler will not load any extra files from disk.
+       If a contract imports a file that is not present in ``sources``, the compiler will use the
+       file loader as in any other situation, which may result in the source being read from disk
+       (or provided by the callback when using the JavaScript interface).
+
+#. **Standard JSON (as URL)**
+
+   When using :ref:`Standard JSON interface <compiler-api>` it is possible to tell the compiler to
+   use the file loader to obtain the content:
+
+   .. code-block:: json
+
+       {
+           "language": "Solidity",
+           "sources": {
+               "/usr/local/dapp-bin/token.sol": {
+                   "urls": ["/projects/mytoken.sol"]
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   The value specified in ``urls`` does not affect the source unit name and is not included in
+   contract metadata.
+   It is only passed to the file loader and used to locate the file.
+
+   As the name of the attribute implies, the value could be an URL if supported by the loader.
+   This may only be the case when using the JavaScript interface with a callback that supports URLs.
+   The default loader only supports paths and will attempt to use the URL as a local path.
+   This will most likely fail and the loader will proceed to the next value on the list.
+
+   When using the default file loader, paths in ``urls`` are affected by :ref:`base path <base-path>`
+   and any other transformations performed by it.
+
+   .. index:: ! import; path
+
+#. **import statement**
+
+   The ``import`` statement requests a module from the compiler and allows to access certain symbols
+   from that module.
+   We will refer to the path used in the statement as *import path*.
+
+   The import path is translated into a source unit name and then the compiler uses the name
+   to look up the file in its virtual filesystem.
+   The are two types of imports, each with different rules for this translation:
+   :ref:`direct imports <direct-imports>` let you specify the full source unit name while in
```

Comment:
Why not "absolute"?

Reply:
Because when I see absolute imports, I think about absolute paths in the real filesystem and I really want to avoid that connotation. Thinking about absolute imports was really giving me the wrong idea about how this works when I was trying to understand this initially. I think we should have a different word to refer to them.